sdk/src/main/java/com/hedera/hashgraph/sdk/Executable.java

```diff
@@ -353,10 +353,13 @@ public O execute(Client client, Duration timeout) throws TimeoutException, Prech
                 throw new MaxAttemptsExceededException(lastException);
             }
 
-            if (Instant.now().isAfter(timeoutTime)) {
+            Duration deadline = Duration.between(Instant.now(), timeoutTime);
+            if (deadline.isNegative() || deadline.isZero()) {
                 throw new TimeoutException();
             }
 
+            this.setGrpcDeadline(deadline);
```

Comment:
I don't think we should modify the `Executable` properties set by the user. Instead, keep it as the user provided value and calculate the min deadline with time remaining and pass to `GrpcRequest`. Otherwise if you modify the original you can't calculate the min against timeout on the next retry.